src/include/ikarus/Assembler/SimpleAssemblers.h

```diff
@@ -23,75 +27,76 @@ namespace Ikarus::Assembler {
   private:
     double& getScalarImpl(Ikarus::FiniteElements::ScalarAffordances scalarAffordances) {
       scal = 0;
-      for (auto [fe, dofs, vars] : dofManager_->elementDofsVariableTuple())
-        scal += calculateScalar(fe, vars, scalarAffordances);
+      for (auto [fe, dofs, vars] : feManager_->elementDofsVariableTuple()) {
+        scal += calculateScalar(fe, scalarAffordances, vars);
+      }
       return scal;
     }
 
   private:
-    DofManagerType* dofManager_;
+    DofManagerType* feManager_;
     double scal{};
   };
 
   template <typename DofManagerType>
   class VectorAssembler {
   public:
     explicit VectorAssembler(DofManagerType& dofManager)
-        : dofManager_{&dofManager}, vec{Eigen::VectorXd::Zero(dofManager_->correctionSize())} {}
+        : feManager_{&dofManager}, vec{Eigen::VectorXd::Zero(feManager_->correctionSize())} {}

```

Comment:
Vorschlag: die Größe von "vec" erst in der Funktion "getVector" bzw "getVectorImpl" festlegen. Dann kann man den VectorAssembler auch nutzen, wenn sich die correctionSize ändert (bei der bisherigen Variante müsste man jedes Mal einen neuen VectorAssembler erzeugen, da man nicht weiß ob sich die Größe ggf geändert hat)

Reply:
.setZero(size) macht automatisch den resize falls die größe nicht stimmt siehe zB getVectorImpl